### PR TITLE
[homematic] Quick fix for Integer values parser

### PR DESCRIPTION
**Pull request by 3yc**
Homematic quick&diry fix for Integer values parser.

Since Homematic is using integer values that are bigger then Java **Integer**, the binding fails to parse it correctly. This is a quick&dirty fix, a code update to user **Long** instead of **Integer** is still needed.

### CODE REVIEW

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ protected Integer toInteger(Object object) {
             return (Integer) object;
         }
         try {
-            return Double.valueOf(ObjectUtils.toString(object)).intValue();
+            return Long.valueOf(Double.valueOf(ObjectUtils.toString(object)).longValue()).intValue();
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java]: This line became really bad imho, double can be used for the same value, I don't suppose any change is needed on this line.

File: addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java [hunk 2]
```diff
@@ -181,7 +181,7 @@ protected Object convertToType(String value) {
         } else if (value.equalsIgnoreCase("false") || value.equalsIgnoreCase("off")) {
             return (Boolean.FALSE);
         } else if (value.matches("(-|\\+)?[0-9]+")) {
-            return (Integer.valueOf(value));
+            return Integer.valueOf((Long.valueOf(value).intValue()));
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/parser/CommonRpcParser.java]: Imho the following should be enough, although it still makes an overflow mess at runtime:

```java
Long.valueOf(value).intValue();
```

**Comment by gerrieg** [comment]
never heard before

> Since Homematic is using integer values that are bigger then Java Integer

@3yc 
Can you give me an example please? Is it a datapoint or a variable? Please post more informations e.g debug log so i can see what is sending which value.

**Comment by 3yc** [comment]
@gerrieg Welcome. For me it happened while discovering. Its the max value definition for a variable. Here we go:

```
2017-08-20 11:28:18.617 [ERROR] [very.HomematicDeviceDiscoveryService] - For input string: "9999999999"
java.lang.NumberFormatException: For input string: "9999999999"
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65) ~[?:?]
	at java.lang.Integer.parseInt(Integer.java:583) [?:?]
	at java.lang.Integer.valueOf(Integer.java:766) [?:?]
	at org.openhab.binding.homematic.internal.communicator.parser.CommonRpcParser.convertToType(CommonRpcParser.java:184) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at org.openhab.binding.homematic.internal.communicator.parser.CcuVariablesAndScriptsParser.parse(CcuVariablesAndScriptsParser.java:50) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at org.openhab.binding.homematic.internal.communicator.CcuGateway.loadVariables(CcuGateway.java:109) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadChannelValues(AbstractHomematicGateway.java:480) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at org.openhab.binding.homematic.internal.communicator.AbstractHomematicGateway.loadAllDeviceMetadata(AbstractHomematicGateway.java:389) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at org.openhab.binding.homematic.discovery.HomematicDeviceDiscoveryService$1.run(HomematicDeviceDiscoveryService.java:110) [206:org.openhab.binding.homematic:2.2.0.201708192106]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]

```

@martinvw 
Its quick&dirty fix, not a solution, and definitely needs more love. For me the right way to fix it is to use Long for all Integer values in binding. It's a bit bigger task, I think, and I needed it quick so here the dirty patch.

**Comment by gerrieg** [comment]
The max value of a variable is undefined, therefore all number variables are floats. Your fix with a long instead of a integer works only in your special case.
But how did you create a integer variable in your CCU? I can't reproduce, they are always floats.

**Comment by gerrieg** [comment]
OK, i got it now ... the variable value is a float, but the CCU sends the min/max metadata values as non float values. So the binding assumes it is a integer. And if the value is bigger than a Java integer value, the exception occurs.
I am going to fix that.

**Comment by 3yc** [comment]
Great! Thanks for the deep-dive.

**Comment by gerrieg** [comment]
fixed with https://github.com/openhab/openhab2-addons/pull/2572

In the meantime, you can set the max value in the CCU to 9999999999.0
This should also work with the current version of the binding.

**Pull request closed without merging**